### PR TITLE
sysctlfix: close systemd config file before triggering reload

### LINKED ISSUE #21916: sysctlfix doesn't properly flush/close the overrides file before restarting systemd-sysctl unit
### Is there an existing issue for this?

- [X] I have searched the existing issues

### What happened?

problem code:
https://github.com/cilium/cilium/blob/master/tools/sysctlfix/main.go

we noticed flakey behavior on some hosts, where the `net.ipv4.conf.all.rp_filter` setting was not properly set to `0` even after the cilium agent init-containers ran. the generated override file seemed to be in place, but the live value for the aforementioned sysctl was still set to `1` and so the cilium health endpoint was timing out. manually setting the live value of the sysctl to `0` resolved the health endpoint issue.

strongly suspect that because the sysctl.d/ override file is not being properly flushed/closed *before* the restart of the `systemd-sysctl` unit, that the new sysctls aren't always being eval'd as intended.

### Cilium Version

1.11.10

### Kernel Version

5.10.32

### Kubernetes Version

1.23.13

### Sysdump

sorry, airgapped system

### Relevant log output

_No response_

### Anything else?

_No response_

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

### PR DESCRIPTION
**Pull request by dylandreimerink**
I accidentally stumbled onto issue again #21916 after it not being on my radar for a while. This PR should fix the issue.

Fixes: #21916

### CODE REVIEW

File: tools/sysctlfix/main.go [hunk 1]
```diff
@@ -139,3 +108,38 @@ func main() {
 
 	fmt.Printf("systemd unit '%s' restarted\n", *sysctlUnit)
 }
+
+func writeConfig(overwritesPath string) error {
+	f, err := os.OpenFile(overwritesPath, os.O_RDWR|os.O_CREATE, 0644)
+	if err != nil {
+		return fmt.Errorf("unable to create cilium sysctl overwrites config: %w\n", err)
+	}
+	defer f.Close()
```
Comment (bimmlerd) [inline tools/sysctlfix/main.go:119]: Oof, this is subtle. Took me a long time to understand what the difference even was 😅 

Given the timing of `f.Close` is crucial, can we comment on it, or even better, call `f.Sync()` explicitly somewhere?

**Review by bimmlerd — APPROVED** [review]
It's fine as is, but comment/explicit `Sync` would make the intention a bit clearer

File: tools/sysctlfix/main.go (hunk 1 above)
Comment (dylandreimerink) [reply tools/sysctlfix/main.go:119]: That can't hurt. :+1:

**Pull request merged**